src/token/xERC4626.sol

```diff
@@ -0,0 +1,98 @@
+// SPDX-License-Identifier: MIT
+// Rewards logic inspired by xERC20 (https://github.com/ZeframLou/playpen/blob/main/src/xERC20.sol)
+
+pragma solidity ^0.8.0;
+
+import "solmate/mixins/ERC4626.sol";
+import "solmate/utils/SafeCastLib.sol";
+
+/** 
+ @title  An xERC4626 Single Staking Contract
+ @author Tribe DAO
+ @notice This contract allows users to autocompound rewards denominated in an underlying reward token. 
+         It is fully compatible with ERC4626 allowing for DeFi composability.
+         It maintains balances using internal accounting to prevent instantaneous changes in the exchange rate.
+         NOTE: an exception is at contract creation, when a reward cycle begins before the first deposit. After the first deposit, exchange rate updates smoothly.
+*/
+abstract contract xERC4626 is ERC4626 {
+    using SafeCastLib for *;
+
+    /// @dev thrown when syncing before cycle ends.
+    error SyncError();
+
+    /// @notice the length of a rewards cycle
+    uint32 public immutable rewardsCycleLength;
+
+    /// @notice the delayed start of the current cycle
+    uint32 public lastSync;
+
+    /// @notice the end of the current cycle
+    uint32 public rewardsCycleEnd;
+
+    /// @notice the amount of rewards distributed in a given cycle
+    uint160 public lastRewardAmount;
+
+    uint256 internal storedTotalAssets;
+
+    constructor(uint32 _rewardsCycleLength) {
+        rewardsCycleLength = _rewardsCycleLength;
+        // seed initial rewardsCycleEnd
+        rewardsCycleEnd = uint32(block.timestamp) / rewardsCycleLength * rewardsCycleLength;
```

Comment:
Why are you dividing by `rewardsCycleLength^2`? This isn't obvious to me, maybe add a comment?
Also is there a possibility of division errors here when the rewardsCycleLength is too large?

Reply:
The division happens first so this is just rounding to the nearest rewardsCycle. I'll clarify in a comment